src/SonarScanner.MSBuild.PreProcessor/Roslyn/RoslynAnalyzerProvider.cs

```diff
@@ -117,14 +119,32 @@ private Plugin[] CreatePlugins()
                 plugin.AddProperty(property.Id, property.Value);
             }
         }
+        // If both legacy and new properties are present, remove the legacy version
+        if (candidates.ContainsKey(string.Format(ServerPropertyFormat, language) + ".dotnet"))
+        {
+            candidates.Remove(LegacyServerPropertyPrefix + language);
+        }
+        if (candidates.ContainsKey(string.Format(ServerPropertyFormat, language) + ".security"))
+        {
+            candidates.Remove(LegacySecurityFrontEndPropertyPrefix);
+        }
         return candidates.Values.Where(x => x.IsValid).ToArray();
     }
 
     private string PluginPropertyPrefix(string propertyId)
     {
-        var parts = propertyId.Split('.');
-        return parts[0] == LegacyServerPropertyPrefix + language || roslynPropertyKeys.Contains(parts[0])
-            ? parts[0]
-            : null;
+        if (propertyId.Contains("."))
```

Comment:
We can avoid double string search
```suggestion
        var index = propertyId.LastIndexOf('.');
        if (index > 0)
```
or (slightly not so nice, but still acceptable)
```
if (propertyId.LastIndexOf('.') is var index and > 0)
```